### PR TITLE
first draft of MultipleTrainEpochsSampler

### PR DESCRIPTION
**Pull request by jnsLs**
still crashes before epoch 20

### CODE REVIEW

**Comment by Maltimore** [comment]
Personally, I do not use this anymore: I now just use `RandomSampler` of pytorch directly, which allows specifying `num_samples`. `num_samples` can be both greater or smaller than the length of the dataset. This has additional benefits:
- `num_samples` doesn't have to be a multiple of the length of the dataset
- specifying `num_samples` to be smaller than the length of typical datasets, one can always leave `num_samples` the same, e.g. 1000. This is beneficial because if the length of the dataset always changes, one has to adapt other parameters, especially the validation interval and the learning rate schedulers, which operate epoch-based.

**Comment by jnsLs** [comment]
Thank you Malte! :)
